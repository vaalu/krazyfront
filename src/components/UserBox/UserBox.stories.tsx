import React from 'react'
import type { Meta, StoryObj } from '@storybook/react';
import { UserBox } from './UserBox'
const tabsMeta : Meta<typeof UserBox>= {
	title: "krazyfront/UserBox",
	component: UserBox,
	tags: [
		"autodocs"
	],
	parameters: {
	},
	argTypes: {
		items: { control: { type: "text" } }
	}

} 
export default tabsMeta
export const DefaultVerticalTabs: Story = {
	parameters: {
	},
	argTypes: {
	},
	render: () => {
		return (<UserBox />)
	}
}
type Story = StoryObj<typeof tabsMeta>
