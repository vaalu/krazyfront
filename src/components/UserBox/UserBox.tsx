import React, { useEffect, useState } from 'react';
import { Button, FormControl, TextField } from '@mui/material';
import { TailSpin } from 'react-loader-spinner'
import { UserOperations } from '../Services';
export const UserBox = (createHandler:any) => {
	const [newUser, setNewUser] = useState({})
	const [isLoading, setLoading] = useState(true)

	const operations = new UserOperations()

	const handleChange = (event:any) => {
		const {name, value} = event.target
		setNewUser({...newUser, [name]:value})
	}
	const handleCreate = async () => {
		setLoading(true)
		console.log('Creating new user: ', newUser)
		await operations.createUser(newUser)
		setLoading(false)
		console.log('Create handler: ', createHandler)
		createHandler.createHandler()
	}
	useEffect(() => { setLoading(false) }, [])
	if (isLoading) {
		return (
		  <TailSpin color="red" radius={"8px"} />
		)
	  } else {
		return (
			<div>
				<FormControl  style={{"padding":"15"}}>
					<div>
						<TextField label="First Name" type='text' color='primary' name='firstName' onChange={handleChange} />&nbsp;
						<TextField label="Last Name" type='text' color='primary' name='lastName' onChange={handleChange} /><div>&nbsp;</div>
					</div>
					<TextField label="Email" type='text' color='primary' name='email' onChange={handleChange} /><div>&nbsp;</div>
					<TextField label="Phone" type='text' color='primary' name='phone' onChange={handleChange} /><div>&nbsp;</div>
					<div>
						<TextField label="Date of Birth" type='date' color='primary' name='dateOfBirth' onChange={handleChange} />&nbsp;
						<TextField label="Status" type='text' color='primary' name='status' onChange={handleChange} /><div>&nbsp;</div>
					</div>
					<TextField label="Address" type='textarea' color='primary' name='address' onChange={handleChange} /><div>&nbsp;</div>
					<div>
					<Button onClick={handleCreate}>Create New User</Button>
					</div>
				</FormControl>
			</div>
		)
	  }
}