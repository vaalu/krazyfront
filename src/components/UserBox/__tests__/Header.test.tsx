import TestRenderer from 'react-test-renderer'
import React from 'react'
import { UserBox } from '../UserBox'

describe('Header', () => {
	it('Headers with levels test', () => {
		const tree = TestRenderer.create(
		<UserBox />
		).toJSON()
		expect(tree).toMatchSnapshot()
	})
})