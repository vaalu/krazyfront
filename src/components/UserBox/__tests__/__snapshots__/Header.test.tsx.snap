// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Headers with levels test 1`] = `
<div
  aria-busy={true}
  aria-label="tail-spin-loading"
  className="sc-beySPh gZmPOF"
  data-testid="tail-spin-loading"
  role="progressbar"
>
  <svg
    data-testid="tail-spin-svg"
    height={80}
    viewBox="0 0 38 38"
    width={80}
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <linearGradient
        id="a"
        x1="8.042%"
        x2="65.682%"
        y1="0%"
        y2="23.865%"
      >
        <stop
          offset="0%"
          stopColor="red"
          stopOpacity="0"
        />
        <stop
          offset="63.146%"
          stopColor="red"
          stopOpacity=".631"
        />
        <stop
          offset="100%"
          stopColor="red"
        />
      </linearGradient>
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
    >
      <g
        transform="translate(1 1)"
      >
        <path
          d="M36 18c0-9.94-8.06-18-18-18"
          id="Oval-2"
          stroke="red"
          strokeWidth={2}
        >
          <animateTransform
            attributeName="transform"
            dur="0.9s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </path>
        <circle
          cx="36"
          cy="18"
          fill="#fff"
          r={8}
        >
          <animateTransform
            attributeName="transform"
            dur="0.9s"
            from="0 18 18"
            repeatCount="indefinite"
            to="360 18 18"
            type="rotate"
          />
        </circle>
      </g>
    </g>
  </svg>
</div>
`;
